internal/controller/pkg/revision/deployment.go

```diff
@@ -116,6 +116,9 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 		s.Annotations = cc.Annotations
 		d.Labels = cc.Labels
 		d.Annotations = cc.Annotations
+		if cc.Spec.ServiceAccountName != nil {
+			s.Name = *cc.Spec.ServiceAccountName
```

Comment:
I agree that this would simplify most cases where people pass a `ServiceAccountName` today. 
I am a bit hesitant due to the following:

- This _could_ be a breaking change for existing setups where a `ServiceAccount` is created separately.
- I don't feel much comfortable API-wise. So far, any configurations there (except labels/annotations, i.e. metadata) were only affecting the pod spec.

@hasheddan what do you think? Any previous discussion on this?

Reply:
I can understand your concern. I have another idea as below to realize my intention without breaking the compatibility or affecting other resources than Pods:

- Add a new label, for example `crossplane.io/service-account`, to be used in a user-managed ServiceAccount, optionally
- Update the [logic to reconcile ClusterRoleBindings](https://github.com/micnncim/crossplane/blob/a1801bd7581e1df13d379347e7ddec2b2a868c3b/internal/controller/rbac/provider/binding/reconciler.go#L173-L184) to create a RoleBinding for a user-managed ServiceAccount alongside with controller-managed ServiceAccounts if it has the above label

In this approach, in addition to resolving your concern, users can choose to both keep the current behavior and have their ServiceAccount be granted a necessary Role.